r/src/arrowExports.cpp

```diff
@@ -270,6 +270,83 @@ RcppExport SEXP _arrow_DictionaryArray__dictionary(SEXP array_sexp){
 }
 #endif
 
+// array.cpp
```

Comment:
I find this file very error prone to read. I'd recommend going selective import route.

```
arrowExports.cpp # with ifdef import of impl/mock depending on ARROW_R_WITH_ARROW
arrowExports_impl.cpp
arrowExports_mock.cpp
```

Don't do this in this PR, open a followup ticket.

Reply:
Not that I would expect anyone to actually read that file (hence it is supposed to be hidden in pull requests, ...) but I've open that issue:  

https://issues.apache.org/jira/browse/ARROW-5627